### ISSUE TITLE
E-Module Mapping Script

### ISSUE THREAD

**Issue by alFrie** [BAMresearch/LebeDigital#117]
The e-module part got separated from the rest of the CPTO. We have a working mapping script for the simple ontology and metadata to be mapped. So on this branch we will write a mapping script for the e-module. Things to do:

- [x] Edit the metadata exctraction script: Hardcode the specimen age to 28 days and add to yaml-file.
- [x] Edit the metadata exctraction script:  Extract the experiment duration ("Zeit") and add to yaml-file.
- [x] Edit the metadata exctraction script: Make the metadata-dictionary keys match the placeholders.
- [x] Map the metadata in the yaml-file to the ontology by replacing placeholders.
- [ ] Write a test.

**Comment by ThiloMuth** [comment]
Test

**Comment by joergfunger** [comment]
@ThiloMuth  if you accept the invitation to the repo, you can review the merge requests, e.g. [this one](https://github.com/BAMresearch/LebeDigital/pull/118)

**Comment by mattheokru** [comment]
No nothing predefined by the Ontology, I will change it.

**Comment by raviapatel** [comment]
> @mattheokru Is this way of spelling ("Modul") within the e-module ontology on purpose/ predefined by the ontology or something like that? It looks german to me. ![image](https://user-images.githubusercontent.com/93983136/218550519-5093f813-7e29-4037-8205-a87df04abfeb.png)
Yes this is just individual and you are right this is german way of doing it I would propose to change it to YoungsModulusTestSpecimen_ or EModulusTestSpeciemen

**Comment by joergfunger** [comment]
Currently, we store the files locally (as you have mentioned with that path), so I would add for now exactly this path. Ultimately, this would have to go to a file server/mongdb/openBIS with a URI, but please talk to @AidaZt  or @ThiloMuth on how we should then reference these files in our KG.

**Comment by AidaZt** [comment]
Andre suggested that we can reference the the link/URL to the file.
Example of an RDF triple:
<http://bam.de/material#experiment01> <http://bam.de/properties#rawdata> <http://bam.de/dataserver/rawdata.csv> .

**Comment by joergfunger** [comment]
But that links is not existing, or how do we intend to store data such that this link is actually a real reference?

**Comment by firmao** [comment]
> But that links is not existing, or how do we intend to store data such that this link is actually a real reference?

Then, we need, at least to talk about a file server, point straight to github raw files, dereferencing URIs providing RDF content, etc.

I suggest we have a short meeting to have an agreement about the best way for us to deal with the raw files.
What about Monday after 3pm?

Best regards,
Andre Valdestilhas

**Comment by firmao** [comment]
> About the Transducer Column: According to the drawio we're expecting an integer: `"$$TransducerColumn_Value$$"^^xsd:integer` The value gets defined within the metadata extraction script of emodule. We talked about saving a list to that key: [1,2,3], giving this result of the mapped onto: `con:Transducer_ a con:MeasuringGauge, owl:NamedIndividual ; ns3:hasPmdUnit ns3:Q56402798 ; mid:has_column_index "[1, 2, 3]"^^xsd:integer .` We have a list of integers instead of an integer. Is that still valid?

The data type expected is an xsd:integer, therefore it's supposed to be an integer number. If you still not sure about the data type, then store as an xsd:string.

**Comment by AidaZt** [comment]
I thinks so, because we either have string or integer as a type and we can't refer it as xsd:list or something. I think for now leave it as xsd:string.

**Comment by firmao** [comment]
if you still need to store a kind of list of values in RDF, there is an example here:
https://stackoverflow.com/questions/29669555/dynamic-array-in-rdf-xml

**Comment by raviapatel** [comment]
> I thinks so, because we either have string or integer as a type and we can't refer it as xsd:list or something? I think for now leave it as xsd:string.

Ok this is also fine for me

**Comment by joergfunger** [comment]
That should be done during the extraction of the metadata. In the final setup, we will have the data all stored in the openBIS system (metadata), and then extracting this information together with the link to the raw data file should happen. Afterwards, the mapping script will just take that information in the metadata.json and replace that value in the ttl file obtained from the diagrams.net ttl template.
